### DISCUSSION TITLE
osu!taiko screen tearing on macOS M1

### DISCUSSION THREAD

**Discussion by nijynot** [ppy/osu#19180] [Q&A] [answered]
### Type

Performance

### Bug description

The applications start tearing every 30 seconds or so when playing osu!taiko.
I've tried all kinds of different graphics settings, with no luck. VSync nor any other frame limiter option fixes this issue.

Makes the game pretty much unplayable.

### Version

2022.716.1-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/9127307/database.log)
[network.log](https://github.com/ppy/osu/files/9127308/network.log)
[performance.log](https://github.com/ppy/osu/files/9127309/performance.log)
[runtime.log](https://github.com/ppy/osu/files/9127310/runtime.log)

**Comment by peppy** [comment, marked as answer]
That doesn't sound right. macOS *forces* vsync on all apps. Have you been playing with developer tools to disable it at a system level? That's basically the only way this can happen.

**Reply by nijynot** [reply]
No, I should not have disabled vsync at a system level.

**Reply by peppy** [reply]
Are you able to show what you're seeing in a video of your screen or a screen capture? Does it happen if you run the game fully windowed?

**Reply by nijynot** [reply]
Yes, happy to. Here's a screen recording of the tearing.
I've tried with windowed, borderless and fullscreen. None seem to fix the tearing.

The tearing comes and goes in intervals of around 30 seconds or so.

https://user-images.githubusercontent.com/16340977/179394716-83e1e074-ec78-4eb3-8a13-c2b6f8f732ff.mov

**Reply by nijynot** [reply]
Here's an example for when the screen tearing is reduced and vanishes for a bit. (But always comes back soon.)

https://user-images.githubusercontent.com/16340977/179394864-fcb35c84-ec73-49a5-8603-a0d269db5651.mov

**Reply by peppy** [reply]
Just to confirm, that doesn't look like tearing but just really low frames. Are you actually seeing tearing locally..?

**Reply by nijynot** [reply]
Yes quite sure I'm seeing tearing locally.
Note that macOS's screen record records at 30fps, so second video should be even smoother than it is in the video.

As for the lower frame rates, I recorded the two videos in two different occasions, so that's also why the first is capped at 60fps while the other goes higher. (First video has vsync on, the second video has unlimited. This doesn't change a thing tho, I get tearing in both occasions.)

**Reply by nijynot** [reply]
And when I notice tearing, the fps itself does not change. So my guess is that it's tearing.

**Reply by peppy** [reply]
@frenzibyte can you repro any tearing? sounds completely impossible to me, and if it's happening it's apple's end.

**Reply by frenzibyte** [reply]
I tried locally on my display with and without ProMotion, and I sense absolutely no tearing, on all window modes (Windowed, Borderless, Fullscreen). It might be best to include a recording from an external device (i.e. phone) just to confirm the screen tearing.

I've also noticed your FPS capped to 60fps with VSync, do you have ProMotion disabled? or is this on a second external monitor?

**Reply by frenzibyte** [reply]
Actually nevermind, just realized you're running on a M1, not an M1 Pro.

**Reply by peppy** [reply]
@nijynot if you can manage it, i'd love to see a video recorded with a camera (if your phone has slow motion that would be even better) just to better understand what's going on here.

**Reply by nijynot** [reply]
https://user-images.githubusercontent.com/16340977/179605561-fe61d23c-fb0b-410c-a5c9-ba093a9b4eef.MOV


https://user-images.githubusercontent.com/16340977/179606222-26b29904-1116-470c-96d0-e99e4420ba73.MOV

**Reply by nijynot** [reply]
First one is where it's behaving weirdly, and the second one is normal.

**Reply by peppy** [reply]
Thanks. There's still no tearing so that's good to know (because it wouldn't make sense at all), but the performance difference is apparently. Will require some further investigation as I don't believe we have seen this happen for anyone else yet. It could be thermal throttling, or some other app running on your PC kicking in after some period of time.

First thought would be to get an app to monitor your system thermals and check if it's getting out of hand.

**Reply by nijynot** [reply]
Gotcha, sounds good!

I see, that makes sense, that it's performance.
I'll look into the thermal throttling and if there's something that sticks out.

One thing weird about it is that the performance issue comes in *very consistent* cycles.
It's about every 30 - 50 seconds or so that the performance drops a lot.
Although, the FPS stays the same, but the rendering on the screen starts to flicker like in the first video.

Either way, I'll check the thermal stuff as a first step, if I can find software to track it etc.

**Reply by peppy** [reply]
You can try https://github.com/exelban/stats

**Reply by nijynot** [reply]
Thanks for the link.
After a lot of triaging, it seems that indeed it is a performance issue, and most likely thermal throttling.
I think that the computer is just barely able to run osu!lazer.
I'm going to keep monitoring and will try manually controlling the fans at some point (which should solve the issue).

That aside, with this I feel the issue is now done and I'll mark it as resolved.
Much thanks for the support.

**Reply by peppy** [reply]
It should be able to run fine, for reference. Try setting osu! to single thread and frame limiter at vsync or 2x refresh.

**Reply by 2minwia** [reply]
chiming in to say i have the same issue as in op's videos on my M1 MBA if more help testing is needed
notably, it tends to happen very consistently when im using an external display?

**Reply by peppy** [reply]
If you can repro with vsync turned on, could you please capture video of it happening with the full performance display visible (Ctrl+F11 twice)?

**Reply by 2minwia** [reply]
got it to repro, hits around when the song transitions in the vid (7 seconds in)

https://user-images.githubusercontent.com/2979581/181077855-9e73e486-dd74-4d52-a582-42eafd827d49.mp4

slowmo-camera-record version (that i had to upload to streamable, because somehow its 2 minutes long) https://streamable.com/oijw69

**Reply by peppy** [reply]
I can't see any issues in either video.
